apps/communications/contacts/js/contacts.js

```diff
@@ -242,26 +239,22 @@ var Contacts = (function() {
   };
 
   var contactListClickHandler = function originalHandler(id) {
-    initDetails(function onDetailsReady() {
-      ContactsService.get(id, function findCb(contact, fbContact) {
 
-        currentContact = contact;
-        currentFbContact = fbContact;
+    if (!ActivityHandler.currentlyHandling) {
+      window.location.href = ParamUtils.generateUrl('detail', {contact:id});
+      return;
+    }
 
-        if (ActivityHandler.currentActivityIsNot(['import'])) {
-          if (ActivityHandler.currentActivityIs(['pick'])) {
-            ActivityHandler.dataPickHandler(currentFbContact || currentContact);
-          }
-          return;
+    ContactsService.get(id, function findCb(contact) {
+      currentContact = contact;
+      if (ActivityHandler.currentActivityIsNot(['import'])) {
+        if (ActivityHandler.currentActivityIs(['pick'])) {
+          ActivityHandler.dataPickHandler(currentContact);
         }
+        return;
+      }
 
-        contactsDetails.render(currentContact, currentFbContact);
-        if (window.Search && Search.isInSearchMode()) {
-          MainNavigation.go('view-contact-details', 'go-deeper-search');
-        } else {
-          MainNavigation.go('view-contact-details', 'go-deeper');
-        }
-      });
+      window.location.href = ParamUtils.generateUrl('detail', {contact:id});
```

Comment:
I'm wondering if this line of code should be executed ever. If we are not handling any activity we should have gone through L243 right?
We could potentially also save some cycles checking that the activity is import or pick before calling the ContactsService